lib/SILOptimizer/Transforms/SILSROA.cpp

```diff
@@ -213,9 +213,8 @@ createAllocas(llvm::SmallVector<AllocStackInst *, 4> &NewAllocations) {
       Optional<SILDebugVariable> NewDebugVarInfo =
           SILDebugVariable::createFromAllocation(AI);
       if (NewDebugVarInfo)
-        // TODO: Handle DIExpr that is already attached
-        NewDebugVarInfo->DIExpr = SILDebugInfoExpression::createFragment(VD);
-
+        NewDebugVarInfo->DIExpr.append(
```

Comment:
I was thinking: what if we don't clear out DIExpr from the existing VarInfo in SILDebugVariable::createFromAllocation so that this line will append fragment of the inner member. But then I realized we can't have more than one fragment inside a single DIExpression, right? If that's the case, then I do think this is the best solution we have now (until we have the infrastructure to merge two fragments inside a DIExpression)

Reply:
You're right — just appending the fragments would be the right thing to do, but it's currently not actually supported. I don't think there's much missing from making this work though.

Reply:
Cool! then I think this LGTM.